packages/puppeteer/src/node/install.ts

```diff
@@ -48,28 +48,90 @@ export async function downloadBrowser(): Promise<void> {
 
   const unresolvedBuildId =
     configuration.browserRevision || PUPPETEER_REVISIONS[product] || 'latest';
+  const unresolvedShellBuildId =
+    configuration.browserRevision ||
+    PUPPETEER_REVISIONS['chrome-headless-shell'] ||
+    'latest';
 
-  const buildId = await resolveBuildId(browser, platform, unresolvedBuildId);
   // TODO: deprecate downloadPath in favour of cacheDirectory.
   const cacheDir = configuration.downloadPath ?? configuration.cacheDirectory!;
 
   try {
-    const result = await install({
-      browser,
-      cacheDir,
-      platform,
-      buildId,
-      downloadProgressCallback: makeProgressCallback(browser, buildId),
-      baseUrl: downloadBaseUrl,
-    });
-
-    logPolitely(
-      `${supportedProducts[product]} (${result.buildId}) downloaded to ${result.path}`
-    );
+    const installationJobs = [];
+
+    if (configuration.skipChromeDownload) {
+      logPolitely('**INFO** Skipping Chrome download as instructed.');
+    } else {
+      const buildId = await resolveBuildId(
+        browser,
+        platform,
+        unresolvedBuildId
+      );
+      installationJobs.push(
+        install({
+          browser,
+          cacheDir,
+          platform,
+          buildId,
+          downloadProgressCallback: makeProgressCallback(browser, buildId),
+          baseUrl: downloadBaseUrl,
+        })
+          .then(result => {
+            logPolitely(
+              `${supportedProducts[product]} (${result.buildId}) downloaded to ${result.path}`
+            );
+          })
+          .catch(error => {
+            throw new Error(
+              `ERROR: Failed to set up ${supportedProducts[product]} v${buildId}! Set "PUPPETEER_SKIP_DOWNLOAD" env variable to skip download.`,
+              {
+                cause: error,
+              }
+            );
+          })
+      );
+    }
+
+    if (browser === Browser.CHROME) {
+      if (configuration.skipChromeHeadlessShellDownload) {
+        logPolitely('**INFO** Skipping Chrome download as instructed.');
+      } else {
+        const shellBuildId =
+          browser === Browser.CHROME
+            ? await resolveBuildId(browser, platform, unresolvedShellBuildId)
+            : '';
```

Comment:
We check for this a few line prior.
```suggestion
        const shellBuildId = await resolveBuildId(browser, platform, unresolvedShellBuildId);
```